File: operator-api/src/routes/os.js
import 'dotenv/config'
import { Hono } from 'hono'

export const osRouter = new Hono()

osRouter.get('/os/locale', async (c) => {
  const locale = process.env.LANG || 'en_US.UTF-8'
  const keyboard_layout = process.env.XKB_DEFAULT_LAYOUT || 'us'
  const timezone = process.env.TZ || 'UTC'
  return c.json({ locale, keyboard_layout, timezone })
})

osRouter.post('/os/locale', async (c) => {
  const body = await c.req.json()
  const locale = body?.locale || process.env.LANG
  const keyboard_layout = body?.keyboard_layout || 'us'
  const timezone = body?.timezone || 'UTC'
  process.env.LANG = locale
  process.env.TZ = timezone
  return c.json({ updated: true, requires_restart: false })
})

File: server-sigma/internal/routes/osinfo/os.go
package osinfo

import (
	"encoding/json"
	"net/http"
	"os"

	"github.com/go-chi/chi/v5"
)

func Router() *chi.Mux {
	r := chi.NewRouter()

	r.Get("/os/locale", func(w http.ResponseWriter, r *http.Request) {
		locale := os.Getenv("LANG")
		if locale == "" {
			locale = "en_US.UTF-8"
		}
		kb := os.Getenv("XKB_DEFAULT_LAYOUT")
		if kb == "" {
			kb = "us"
		}
		tz := os.Getenv("TZ")
		if tz == "" {
			tz = "UTC"
		}
		json.NewEncoder(w).Encode(map[string]any{"locale": locale, "keyboard_layout": kb, "timezone": tz})
	})

	r.Post("/os/locale", func(w http.ResponseWriter, r *http.Request) {
		var b struct {
			Locale         string `json:"locale"`
			KeyboardLayout string `json:"keyboard_layout"`
			Timezone       string `json:"timezone"`
		}
		_ = json.NewDecoder(r.Body).Decode(&b)
		if b.Locale != "" {
			os.Setenv("LANG", b.Locale)
		}
		if b.Timezone != "" {
			os.Setenv("TZ", b.Timezone)
		}
		json.NewEncoder(w).Encode(map[string]any{"updated": true, "requires_restart": false})
	})

	return r
}